lnwallet/channel.go

```diff
@@ -1583,6 +1587,10 @@ func (lc *LightningChannel) remoteLogUpdateToPayDesc(logUpdate *channeldb.LogUpd
 	case *lnwire.UpdateFailHTLC:
 		ogHTLC := localUpdateLog.lookupHtlc(wireMsg.ID)
 
+		if ogHTLC.addCommitHeightLocal == 0 {
+			ogHTLC.addCommitHeightLocal = commitHeight - 1
```

Comment:
Can't we rather set the height when the `ogHTLC` is added to the `localUpdateLog`?

Reply:
It would normally be set here:
https://github.com/lightningnetwork/lnd/blob/02f8a0db2b409bc394c911b233ebf7a1e01e14be/lnwallet/channel.go#L1815-L1826

but it's been removed from the local commitment and so `addCommitHeightLocal` isn't set. Opted to set it here instead and defer overhaul of `restoreStateLogs` until a bigger refactor.

Reply:
> but it's been removed from the local commitment in this specific case

What has been removed?

Reply:
Updates in `remoteLogUpdateToPayDesc` are settles/fails/malformed fails/fee updates that the remote has sent us:
```
<---fail---
<---sig---
----rev--->
```

Once we send the revocation (we are on the LHS), the original HTLC we sent out is no longer on our commitment. Thus `outgoingLocalAddHeights[htlc.HtlcIndex]` is 0 and so `htlc.addCommitHeightLocal` is also 0.

Reply:
Aha, I see! Wouldn't it be more natural to set the `htlc.addCommitHeightLocal = localCommitment.Height-1` here then, before restoring it? The same for incoming HTLCs.

Reply:
It is now set in the `restoreStateLogs` function instead. Opted to set it to `localCommitment.height` and left out the `-1`.